code/modules/admin/tickets/admintickets.dm

```diff
@@ -0,0 +1,183 @@
+/*
+Admin ticket system by Birdtalon
+*/
+//Global holder
+
+var/global/datum/adminTicketHolder/globAdminTicketHolder = new /datum/adminTicketHolder
+
+//Defines
+//Deciseconds until ticket becomes stale if unanswered. Alerts admins.
+#define ADMIN_TICKET_TIMEOUT 6000 // 10 minutes
+//Decisecions before the user is allowed to open another ticket while their existing one is open.
+#define ADMIN_TICKET_DUPLICATE_COOLDOWN 3000 // 5 minutes
+
+//Status defines
+#define ADMIN_TICKET_OPEN       1
+#define ADMIN_TICKET_CLOSED     2
+#define ADMIN_TICKET_RESOLVED   3
+#define ADMIN_TICKET_STALE      4
+
+//Datum holding all tickets
+/datum/adminTicketHolder
+	var/ticketCounter = 1 // Counts the tickets and used to assign the id number
+	var/list/allTickets = list()
+
+//Return the current ticket number ready to be called off.
+/datum/adminTicketHolder/proc/getTicketCounter()
+	return ticketCounter
+
+//Return the ticket counter and increment
+/datum/adminTicketHolder/proc/getTicketCounterAndInc()
+	. = ticketCounter
+	ticketCounter++
+	return
+
+/datum/adminTicketHolder/proc/resolveAllOpenTickets() // Resolve all open tickets
+	for(var/i in allTickets)
+		var/datum/admin_ticket/T = i
+		resolveTicket(T.ticketNum)
+
+//Open a new ticket and populate details then add to the list of open tickets
+/datum/adminTicketHolder/proc/newTicket(var/client/C, var/passedContent, var/title)
+	if(!C || !passedContent)
+		return
+
+  //Check if the user has an open ticket already within the cooldown period, if so we don't create a new one and re-set the cooldown period
+	var/datum/admin_ticket/existingTicket = checkForOpenTicket(C)
+	if(existingTicket)
+		existingTicket.setCooldownPeriod()
+		to_chat(C, "<span class='adminticket'>Your ticket #[existingTicket.ticketNum] remains open! Visit \"My tickets\" under the Admin Tab to view it.</span>")
+		return
+
+	if(!title)
+		title = passedContent
+
+	var/datum/admin_ticket/T =  new /datum/admin_ticket
+	T.clientName = C
+	T.timeOpened = worldtime2text()
+	T.title = title
+	T.content += passedContent
+	T.locationSent = C.mob.loc.loc.name
+	T.mobControlled = C.mob
+	T.ticketState = ADMIN_TICKET_OPEN
+	T.timeUntilStale = world.time + ADMIN_TICKET_TIMEOUT
+	T.setCooldownPeriod()
+	T.ticketNum = getTicketCounterAndInc()
+	allTickets += T
+
+	//Inform the user that they have opened a ticket
+	to_chat(C, "<span class='adminticket'>You have opened admin ticket number #[(globAdminTicketHolder.getTicketCounter() - 1)]! Please be patient and we will help you soon!</span>")
+
+	//Begin the stale count for this ticket.
+	spawn()
+	T.beginStaleCount()
+
+//Set ticket state with key N to open
+/datum/adminTicketHolder/proc/openTicket(var/N as num)
```

Comment:
All of these `as num` are useless and should be removed- it's only used for verbs.